Makefile

```diff
@@ -61,6 +67,7 @@ AFLAGS = -fsanitize=address #-fsanitize=undefined -fno-omit-frame-pointer
 
 OPT    = -O -g -march=native -ffp-contract=fast -fopenmp-simd
 CFLAGS = -std=c99 $(OPT) -Wall -Wextra -Wno-unused-parameter -fPIC -MMD -MP
+CXXFLAGS = $(OPT) -Wall -Wextra -Wno-unused-parameter -fPIC 
```

Comment:
Can we include `-MMD` here so that we get automatic dependencies?

Reply:
Should I also add the `-MP` flag?

Reply:
Yes, please.